soc/riscv/riscv-privilege/hpmicro/sections.ld

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+#include <zephyr/linker/linker-tool.h>
+#include <zephyr/devicetree.h>
+
+#ifdef CONFIG_XIP
+__nor_cfg_option_load_addr__ = DT_REG_ADDR(DT_CHOSEN(zephyr_flash)) + 0x400;
+__boot_header_load_addr__ = DT_REG_ADDR(DT_CHOSEN(zephyr_flash)) + 0x1000;
+__app_load_addr__ = DT_REG_ADDR(DT_CHOSEN(zephyr_flash)) + 0x3000;
+__boot_header_length__ = __boot_header_end__ - __boot_header_start__;
+__app_offset__ = __app_load_addr__ - __boot_header_load_addr__;
+#endif
+
+SECTIONS
+{
+#ifdef CONFIG_XIP
+	.nor_cfg_option __nor_cfg_option_load_addr__ : {
+		KEEP(*(.nor_cfg_option))
+	} GROUP_LINK_IN(MCU_BOOT)
+
+	.boot_header __boot_header_load_addr__ : {
+		__boot_header_start__ = .;
+		KEEP(*(.boot_header))
+		KEEP(*(.fw_info_table))
+		KEEP(*(.dc_info))
+		__boot_header_end__ = .;
+	} GROUP_LINK_IN(MCU_BOOT)
+#endif
```

Comment:
mcuboot stuff here? @nordicjm

Reply:
> mcuboot stuff here? @nordicjm

XPI NOR boot flash configuration option
It reads this field during the XPI NOR boot process to decide whether to start the firmware

There are comments on this part of the content in the following files
https://github.com/zephyrproject-rtos/zephyr/pull/46329/files#diff-fbd6285e9b4957ff9b98ae797c036a3b42d15ac84ff00b34973fc4d6a78f85bb